joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -64,85 +64,91 @@ def init_collada(self, robot):
         for child in robot.childNodes:
             if child.nodeType is child.TEXT_NODE:
                 continue
-            if child.localName == 'joint':
-                name = child.getAttribute('name')
-                if child.getElementsByTagName('revolute'):
-                    joint = child.getElementsByTagName('revolute')[0]
-                else:
-                    self.get_logger().warn('Unknown joint type %s', child)
-                    continue
+            if child.localName != 'joint':
+                continue
+            name = child.getAttribute('name')
+            if child.getElementsByTagName('revolute'):
+                joint = child.getElementsByTagName('revolute')[0]
+            else:
+                self.get_logger().warn('Unknown joint type %s', child)
+                continue
 
-                if joint:
-                    limit = joint.getElementsByTagName('limits')[0]
+            if joint:
+                limit_list = joint.getElementsByTagName('limits')
+                if limit_list:
```

Comment:
This is again changing the semantics slightly.  I think here we should probably do something more like:

```suggestion
                if not limit_list:
                    continue
```

Reply:
Done